lib/cc/analyzer/source_extractor.rb

```diff
@@ -17,7 +19,14 @@ def extract(location)
 
       private
 
-      attr_reader :location, :source
+      attr_reader :source
+
+      def validate_location(location)
+        validator = IssueValidations::LocationFormatValidation::Validator.new(location)
+        unless validator.valid?
+          raise InvalidLocation
```

Comment:
will this exception have information about why its invalid?  is it just `Location is not formatted correctly`?  i think the more detail the better.